docs/style-spec/expressions.md

```diff
@@ -0,0 +1,40 @@
+# Style Function Expressions
+
+**Constants:**
+- LN2
+- PI
+- e
```

Comment:
> Better to keep the grammar simple and use s-expressions as the only syntactical element.

By the way, one could make the same argument for treating bare strings in the JSON as constant values always and requiring the use of a `{ref: …}` object to represent a property reference. But that would either move the expression syntax away from the filter syntax or necessitate a backwards-incompatible change to the filter syntax.

NSExpression format strings make a distinction between `keyPath` and `'string'` or `"string"`, which makes it possible to represent the style JSON `["==" "property", "string"]` as either `property = 'string'` or `'string' = property` in an NSExpression format string.